src/hypothesis/searchstrategy/misc.py

```diff
@@ -77,7 +77,7 @@ class SampledFromStrategy(SearchStrategy):
 
     def __init__(self, elements):
         SearchStrategy.__init__(self)
-        self.elements = tuple(elements)
+        self.elements = d.check_sample(elements)
```

Comment:
Yes, but it wasn't previously converted to a tuple - right now you can try choosing from a set and just get an IndexError when it eventually tries to pick one.  Moving the deprecation logic to check_sample will take care of that too though.